src/ReactAdmin.php

```diff
@@ -15,7 +58,44 @@ public function getTemplates(): array
 
     protected function getDefaultExtraClasses(): array
     {
-        return ['react-admin'];
+        return ['react-admin', 'cms-content'];
+    }
+
+    /**
+     * @return HTTPResponse
+     */
+    public function notfound(HTTPRequest $request)
+    {
+        // This just tries to return a 404 when the route is not found but still let the front end handle it
+        return $this->index($request)->setStatusCode(404);
+    }
+
+    /**
+     * @param HTTPRequest $request
+     * @return HTTPResponse
+     */
+    public function index($request)
+    {
+        $bootstrapConfig = [
+            'configKey' => static::class,
+            'component' => $this->getComponent(),
+            'props' => $this->getProps(),
+        ];
+        Requirements::customScript(
+            'var reactAdminConfig = ' .
+            json_encode($bootstrapConfig) .
+            ';',
+        );
+        return parent::index($request);
+    }
```

Comment:
Instead of passing props and component names via data attr on an HTML tag, we define a global scope JS variable called `reactAdminConfig`. Our bootstrapping logic just reads it from there.